packages/sdk/contractkit/src/wrappers/SortedOracles.test.ts

```diff
@@ -35,252 +40,432 @@ testWithGanache('SortedOracles Wrapper', (web3) => {
-        const initialRates: OracleRate[] = await sortedOracles.getRates(CeloContract.StableToken)
+  // NOTE: These values are set in test-utils/network-config.json, and are derived
+  // from the MNEMONIC. If the MNEMONIC has changed, these will need to be reset.
+  // To do that, look at the output of web3.eth.getAccounts(), and pick a few
+  // addresses from that set to be oracles
+  const stableTokenOracles: Address[] = NetworkConfig.stableToken.oracles
+  // Use same oracle addresses for CELO/BTC as well
+  const celoBtcOracles: Address[] = NetworkConfig.stableToken.oracles
+  const oracleAddress = stableTokenOracles[stableTokenOracles.length - 1]
 
-        const tx = await sortedOracles.report(CeloContract.StableToken, value, oracleAddress)
-        await tx.sendAndWaitForReceipt()
+  const celoBtcPair = newPairWithIdentifier('CELO/BTC')
+  let stableTokenSortedOracles: SortedOraclesWrapper
+  let btcSortedOracles: SortedOraclesWrapper
 
-        const resultingRates: OracleRate[] = await sortedOracles.getRates(CeloContract.StableToken)
-        expect(resultingRates).not.toMatchObject(initialRates)
-      })
+  let allAccounts: Address[]
+  let stableTokenAddress: Address
+  let nonOracleAddress: Address
+  let btcOracleOwner: Address
 
-      describe('when inserting into the middle of the existing rates', () => {
-        beforeEach(async () => {
-          const rates = [15, 20, 17]
-          await reportAsOracles(stableTokenOracles, rates)
-        })
+  beforeAll(async () => {
+    allAccounts = await web3.eth.getAccounts()
 
-        const expectedLesserKey = stableTokenOracles[0]
-        const expectedGreaterKey = stableTokenOracles[2]
+    btcOracleOwner = allAccounts[0]
+    btcSortedOracles = await newSortedOracles(btcOracleOwner)
+    stableTokenSortedOracles = await kit.contracts.getSortedOracles()
+    stableTokenAddress = await kit.registry.addressFor(CeloContract.StableToken)
 
-        const expectedOracleOrder = [
-          stableTokenOracles[1],
-          stableTokenOracles[2],
-          oracleAddress,
-          stableTokenOracles[0],
-        ]
+    // For StableToken the oracles are setup in migrations, for our
+    // custom CELO/BTC oracle we need to add them manually
+    for (const oracle of celoBtcOracles) {
+      await addOracleForPair(btcSortedOracles, celoBtcPair, oracle, btcOracleOwner)
+    }
 
-        it('passes the correct lesserKey and greaterKey as args', async () => {
-          const tx = await sortedOracles.report(CeloContract.StableToken, value, oracleAddress)
-          const actualArgs = tx.txo.arguments
-          expect(actualArgs[2]).toEqual(expectedLesserKey)
-          expect(actualArgs[3]).toEqual(expectedGreaterKey)
+    nonOracleAddress = allAccounts.find((addr) => {
+      return !stableTokenOracles.includes(addr)
+    })!
+  })
 
-          await tx.sendAndWaitForReceipt()
-        })
+  describe('reporting for the cUSD StableToken', () => {
```

Comment:
Since the tests are basically the same for both cUSD and CELO/BTC, you could simplify this file with something like:

```typescript
const sortedOraclesAndIdentifiers = [
  [CeloContract.StableToken, stableTokenSortedOracles],
  [celoBtcPair, btcSortedOracles]
]

for (const [identifier, sortedOracles] of sortedOraclesAndIdentifiers) {
   // All the duplicated tests
}

```

Reply:
Yeah I thought about something like this, the only reason I didn't was that the test logic that's being tested doesn't actually change between using a PairWithidentifier vs using the CeloToken enum, so it felt like I'm just adding tests and thus bloat to the CI without actually increasing the test utility. So I cherry-picked the tests in order to just touch calling the wrapper methods with the new type so that I ensure that the interface works, and only test some advanced functionality in one of the blocks.

Reply:
That being said I'm also ok with doing all tests for both cases.

Reply:
Good point. My philosophy skews towards overkill when it comes to tests. I think it's fine to do a smaller set of them.